packages/core/src/login/webview/vue/amazonq/backend_amazonq.ts

```diff
@@ -58,6 +60,49 @@ export class AmazonQLoginWebview extends CommonAuthWebview {
         return AuthUtil.instance.findUsableQConnection(connections)
     }
 
+    /**
+     * Attempts to re-use a valid connection in AWS Toolkit.
+     * This logic should only apply to below users:
+     * 1. They have a invalid builder id connection in Amazon Q that do not have all 5 scopes
+     * For these users, if there is a valid builder id connection in AWS Toolkit,
+     * We attempt to re-use this valid builder id connection.
+     */
+    async tryReuseConnectionFromAwsToolkit() {
```

Comment:
Not related to this specifically but can you add a test in `techdebt.test.ts` to revert this work at a certain date.